packages/salmon-wallet/src/SalmonWalletMnemonic.ts

```diff
@@ -0,0 +1,36 @@
+import { Network } from '@defichain/jellyfish-network'
+import { JellyfishWallet, WalletAccountProvider } from '@defichain/jellyfish-wallet'
+import { Bip32Options, MnemonicHdNode, MnemonicHdNodeProvider } from '@defichain/jellyfish-wallet-mnemonic'
+import { WhaleWalletAccount } from '@defichain/whale-api-wallet'
+
+const PURPOSE_LIGHT_ORACLE_ADAPTER = 2
+
+/**
+ * Basic wrapper around JellyfishWallet, so we can extend with any oracle specific
+ * functionality we might need
+ */
+
+export class SalmonWalletMnemonic extends JellyfishWallet<WhaleWalletAccount, MnemonicHdNode> {
+  constructor (
+    words: string[],
+    network: Network,
+    accountProvider: WalletAccountProvider<WhaleWalletAccount>
+  ) {
+    const hdNodeProvider = MnemonicHdNodeProvider.fromWords(words,
+      SalmonWalletMnemonic.getBip32Options(network))
+
+    const coinType: number = JellyfishWallet.COIN_TYPE_DFI
+    const purpose: number = PURPOSE_LIGHT_ORACLE_ADAPTER
+    super(hdNodeProvider, accountProvider, coinType, purpose)
```

Comment:
Since this "purpose" isn't reserved in the jellyfish package, I feel like there is no point explicitly having a purpose just for it? As it doesn't require any functionality beyond what the default offers? We might be able to just use the default constant directly.

> I am on the fence as I am writing this, what do you think?